### ISSUE TITLE
Copy Engine without Connect to DB

### ISSUE THREAD

**Issue by micanzhang** [go-xorm/xorm#705]
for make logger associated with request context, we want to log with request info like trace id; but i 
found `core.ILogger` as field of `xorm.Engine`, my first idea is to `NewEngine` or `engine.Clone()`
to let every request has its own engine; the problem is `NewEngine` always build a new connection with DB server, so my question is it possible add a new method `engine.Copy()` which reuse  `core.DB` ? or is there a better solution to do that?

**Comment by lunny** [comment]
core.ILogger is an interface you can implement your own logger.
Maybe you need is
```go
sess := engine.NewSession()
defer sess.Close()
```
engine will keep only 1 core.DB and all session could share it connection pool. One session only used by only 1 goroutine.

**Comment by micanzhang** [comment]
yes, i know that, question is i wanna use `xorm.ShowSQL(true)`  and `xorm.ShowExecTime(true)`
to log sql execution info and request info,  the `ILogger` interface doesn't have ability to log trace info which depend on request
